### PR TITLE
Added optional skipping of adding dataset in onboarding

### LINKED ISSUE #3972: Public datasets are missing form onboarding
User feedback:

> While creating my organisation on webknossos.org I didn't see the full flight brain datasets. Would be nice if this was a default test dataset

On wk.org, when using the onboarding flow and creating a new organization you are prompted to upload a dataset or download a test datasets. The public datasets on the wk instance are ignored / not shown.

We should feature the public datasets more prominently in this step. Perhaps, have a list of the first 10 public datasets and "see more" or so.

image

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR makes it optional whether the onboarding includes the prompt to upload an own dataset. This is done by setting the `skipAddingDatasetsInOnboarding` in `application.conf` to `true`.
### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- Set the `skipAddingDatasetsInOnboarding` feature flag to true.
- Logout and got to `<base-url>/onboarding`.
- The onboarding should no longer include the third page that asks the user to upload her/his own dataset.

### Issues:
- fixes #3972

------
- ~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~
- ~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~
- ~~[ ] Updated [documentation](../blob/master/docs) if applicable~~
- ~~[ ] Adapted [wk-connect](https://github.com/scalableminds/webknossos-connect) if datastore API changes~~
- ~~[ ] Needs datastore update after deployment~~
- [x] Ready for review

### CODE REVIEW

File: frontend/javascripts/admin/onboarding.js [hunk 1]
```diff
@@ -475,14 +476,15 @@ class OnboardingView extends React.PureComponent<Props, State> {
...
+    const skipAddingDatasets = features().skipAddingDatasetsInOnboarding;
     const currentStepContent = (() => {
       switch (this.state.currentStep) {
         case 0:
           return this.renderCreateOrganization();
         case 1:
           return this.renderCreateAccount();
         case 2:
-          return this.renderUploadDatasets();
+          return skipAddingDatasets ? this.renderWhatsNext() : this.renderUploadDatasets();
```
Comment (philippotto) [inline frontend/javascripts/admin/onboarding.js]: I'd suggest to have a method like this:

```
getAvailableSteps() {
  if (demoInstance) {
  	return [
		["Create Organization", this.renderCreateOrganization],
		["Create Account", this.renderCreateAccount],
		["What's Next?", this.renderWhatsNext],
  	]
  } else {
  	return [
		["Create Organization", this.renderCreateOrganization],
		["Create Account", this.renderCreateAccount],
		["Add Dataset", this.renderUploadDatasets],
		["What's Next?", this.renderWhatsNext],
  	]
  }
}
```

Then, this method can be used for `currentStepContent` and when defining the children of `<Step>`. For that, the render methods need to be changed to `render... = () => {}` to fix the binding.

Comment (MichaelBuessemeyer) [reply frontend/javascripts/admin/onboarding.js]: Nice suggestion! I took it a bit further and used objects inside the array with the structure 
```{ title:"...", component: some_function }```.
 I think this makes the code that uses this function's returned array a bit easier to understand :slightly_smiling_face:

Comment (philippotto) [reply frontend/javascripts/admin/onboarding.js]: Awesome! :)

Comment (philippotto) [reply frontend/javascripts/admin/onboarding.js]: > For that, the render methods need to be changed to `render... = () => {}` to fix the binding.

@MichaelBuessemeyer I fixed this in #4067 now :) Please make sure to always test your changes, especially when you change something after the PR was already approved (and thus tested by the reviewer).

Comment (MichaelBuessemeyer) [reply frontend/javascripts/admin/onboarding.js]: Ok sorry, looks like I forgot that 🙈.

**Pull request merged**